### ISSUE TITLE
`torch.nn.functional.interpolate` parameter size for `scale_factor`

### ISSUE THREAD

**Issue by dogeplusplus** [pytorch/pytorch#80786]
### 📚 The doc issue

The documentation for `scale_factor` in `torch.nn.functional.interpolate` states that if a tuple the number of dimensions should be equal to `input.dim()`.

https://github.com/pytorch/pytorch/blob/5953fd9133c0bdcc0158acf1472fac403bc5f636/torch/nn/functional.py#L3774-L3775

However further down the source code the dim is set as the number of spatial dimensions excluding batch/channels.

https://github.com/pytorch/pytorch/blob/5953fd9133c0bdcc0158acf1472fac403bc5f636/torch/nn/functional.py#L3839

which is then checked when passed a tuple or list for `scale_factor`. This means even if `input.dim() == len(scale_factor)` it will still raise.

https://github.com/pytorch/pytorch/blob/5953fd9133c0bdcc0158acf1472fac403bc5f636/torch/nn/functional.py#L3865-L3873

### Suggest a potential alternative/fix

In the docstring we could say `If scale_factor is a tuple, its length has to match the number of spatial dimensions; input.dim() - 2`.

cc @svekars @holly1238
